builder/dockerfile/builder.go

```diff
@@ -100,10 +85,8 @@ func (bm *BuildManager) Build(ctx context.Context, config backend.BuildConfig) (
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	if src, err := bm.initializeClientSession(ctx, cancel, config.Options); err != nil {
-		return nil, err
-	} else if src != nil {
-		source = src
+	if config.Options.SessionID != "" {
```

Comment:
Could we move this check to the API, and either _remove_ or _deprecate_ the `SessionID` field; https://github.com/moby/moby/blob/827cb09f87964ed38b46502f22a585f2ed4a78e1/api/types/client.go#L182


Looks like this field is set in a couple of places in the API, so perhaps we can catch it there; https://github.com/moby/moby/blob/ad1b781e44fa1e44b9e654e5078929aec56aed66/api/server/router/build/build_routes.go#L143

Reply:
It's debatable where in the code to put the error. If I put it in the API and remove it from the builder package, then whoever uses the builder package will not have an error when they should. I don't mind that the HTTP handler code doesn't error out and leaves it to the builder package to error out. Also, as you realized later, the SessionID field is still needed for BuildKit, so I'm not planning to remove or deprecate that field.